architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,291 @@
+
+**Description:** A malicious actor could circumvent Kubernetes RBAC controls and gain unauthorized access to Kubernetes secrets managed by Radius. These secrets may contain sensitive information, such as credentials intended for use by applications.
+
+**Impact:** A malicious actor could gain access to sensitive information.
+
+**Mitigations:**
+
+1. Secret data managed by the UCP is stored at rest in Kubernetes secrets or etcd. Our threat model assumes that the API server and mechanisms like Kubernetes authentication/RBAC are not compromised.
+2. Secrets managed by Radius are always placed in the same namespace as the object that "owns" them. This is a requirement of the Kubernetes RBAC model.
+3. Secrets managed by Radius are subject to the Kubernetes RBAC model for controlling access. Operators are expected to limit access for users using existing tools.
+
+**Status:** All mitigations listed are currently active. Operators are expected to secure their cluster and limit access for users.
+
+#### Threat: Escalation of Privilege by Using Radius to Circumvent Kubernetes RBAC Controls
+
+**Description:** A malicious actor could circumvent Kubernetes RBAC controls and create arbitrary resources in Kubernetes by exploiting the Universal Control Plane (UCP). The UCP has the following permissions as of November 24, 2024:
+
+- Create, Delete, Get, List, Patch, Update, and Watch on ConfigMaps, Secrets, Services, Deployments, StatefulSets, and `ucp.dev` resources.
+- All permissions on `api.ucp.dev` resources.
+
+**Impact:** This level of access could allow an attacker to perform unauthorized actions, potentially compromising the entire Kubernetes cluster and its resources. Unauthorized resource creation, modification, or deletion could lead to data breaches, service disruptions, and loss of control over the cluster.
+
+**Mitigations:**
+
+1. **Limit Direct Access to the Radius API Using Kubernetes RBAC:**
+   1. Action: Implement strict RBAC policies to limit which users and service accounts can access the Radius API. Ensure that only trusted and necessary entities have the required permissions.
+   2. Implementation: Review and update the RBAC policies regularly to ensure they adhere to the principle of least privilege.
+2. **Enhance Authorization Controls:**
+   1. Action: Develop and implement more granular authorization controls within the UCP to enforce fine-grained access policies.
+   2. Implementation: Introduce role-based access controls within the UCP itself, allowing for more specific permissions and roles tailored to different user needs.
+
+**Status:** These mitigations are partial and require configuration by the operator. We will revisit and improve this area in the future.
+
+#### Threat: Lack of Role-Based Access Control (RBAC) and Unauthorized Traffic
+
+**Description:** As mentioned above, as of November 15, 2024, the UCP does not implement RBAC, and communication between the UCP and resource providers is unauthenticated.
```

Comment:
This is making two points, and they appear unrelated:

- UCP does not implement RBAC
- communication between the UCP and resource providers is unauthenticated

Reply:
Can you clarify the threat? 

> A malicious actor could do X to accomplish Y

Reply:
Updated the threats with more information and updated the titles.